packages/tasks/src/tasks/text-generation/inference.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Inference code generated from the JSON schema spec in ./spec
+ *
+ * Using src/scripts/inference-codegen
+ */
+
+/**
+ * Inputs for Text Generation inference
+ */
+export interface TextGenerationInput {
+	/**
+	 * The text to initialize generation with
+	 */
+	inputs: string[] | string;
+	/**
+	 * Additional inference parameters
+	 */
+	parameters?: TextGenerationParameters;
+	[property: string]: unknown;
+}
+
+/**
+ * Additional inference parameters
+ *
+ * Additional inference parameters for Text Generation
+ */
+export interface TextGenerationParameters {
```

Comment:
TGI also have a `details: bool` parameters to return more details from the process. It also affects the output type.

At the moment we have quite a hacky way to handle both TGI and non-TGI servers in `huggingface_hub.InferenceClient`. What I do is to send TGI-specific parameters to the server and if I receive an error, I retry without the TGI-specific parameters. See [here](https://github.com/huggingface/huggingface_hub/blob/7bb8308909d0981ec911d48a52c16b7ba2c595fc/src/huggingface_hub/inference/_client.py#L1484) on how it's done.

Having a unified API specification should help here. I think it makes sense to have as much parameters as possible in the spec (or at least have the ones for TGI) and if the server does not support them, it ignores them (maybe send back a warning in the output?)